modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/node/NodeServiceImpl.java

```diff
@@ -632,14 +633,14 @@ public Node move( final MoveNodeParams params )
             build().
             execute();
 
-        if ( moveNodeResult.getTargetNode() != null )
+        if ( !moveNodeResult.getMovedNodes().isEmpty() )
         {
-            this.eventPublisher.publish( NodeEvents.moved( moveNodeResult.getSourceNode(), moveNodeResult.getTargetNode() ) );
-            return moveNodeResult.getTargetNode();
+            this.eventPublisher.publish( NodeEvents.moved( moveNodeResult ) );
+            return moveNodeResult.getMovedNodes().get( 0 ).getNode();
         }
         else
```

Comment:
Is this case necessary? If so, it should be tested

Reply:
tested